### PR TITLE
Convert state to use flow

### PR DESCRIPTION
**Pull request by karlenDimla**
Task/Issue URL: https://app.asana.com/0/488551667048375/1206875694725593/f

### Description
See attached task description

### Steps to test this PR
https://app.asana.com/0/0/1206886319244643/f

### CODE REVIEW

**Comment by karlenDimla** [comment]
* **#4334** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4334?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karlenDimla and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: network-protection/network-protection-subscription-internal/src/main/java/com/duckduckgo/networkprotection/subscription/NetpSubscriptionManager.kt [hunk 1]
```diff
@@ -44,4 +50,8 @@ class RealNetpSubscriptionManager @Inject constructor(
         val entitlements = subscriptions.getEntitlementStatus().firstOrNull()
         return@withContext (entitlements?.contains(NetP) == true)
     }
+
+    override fun hasValidEntitlementFlow(): Flow<Boolean> = subscriptions.getEntitlementStatus().map { it.contains(NetP) }
+
+    override suspend fun isTreated(): Boolean = subscriptions.isEnabled()
```
Comment (aitorvs) [inline network-protection/network-protection-subscription-internal/src/main/java/com/duckduckgo/networkprotection/subscription/NetpSubscriptionManager.kt:56]: Why did we move the `fun isTreated` from implementation detail of `NetworkProtectionAccessState` to a more public interface that only uses `NetworkProtectionAccessState` anyways?

Comment (karlenDimla) [reply network-protection/network-protection-subscription-internal/src/main/java/com/duckduckgo/networkprotection/subscription/NetpSubscriptionManager.kt:56]: My intention was to use NetpSubscriptionManager.kt a a wrapper for the subscription API in the Netp land. However, i didn’t move other direct usages of subscription.isEnabled() to this in this PR. I’ll revert this part for now.

File: network-protection/network-protection-subscription-internal/src/main/java/com/duckduckgo/networkprotection/subscription/NetworkProtectionAccessState.kt [hunk 2]
```diff
@@ -83,6 +93,22 @@ class NetworkProtectionAccessState @Inject constructor(
...
+                if (!it) {
+                    // if entitlement check succeeded and no entitlement, reset state and hide access.
+                    handleRevokedVPNState()
+                    NotUnlocked
+                } else {
+                    InBeta(netPWaitlistRepository.didAcceptWaitlistTerms())
+                }
+            }
+        } else {
+            emptyFlow()
```
[the remark this answers is not in the document]
Comment (karlenDimla) [reply network-protection/network-protection-subscription-internal/src/main/java/com/duckduckgo/networkprotection/subscription/NetworkProtectionAccessState.kt]: Fixed

**Review by aitorvs — APPROVED** [review]
LGTM! left some comments, all tests passed

**Pull request merged**